apps/transactions/views.py

```diff
@@ -74,3 +74,20 @@ def perform_create(self, serializer):
 
     def post(self, request, *args, **kwargs):
         return self.create(request, *args, **kwargs)
+
+
+class TransactionsListView(ListAPIView):
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = TransactionSerializer
+
+    def get_queryset(self):
+        user = self.request.user
+        queryset = user.transactions.all()
+        start_date = self.request.query_params.get('start_date')
```

Comment:
https://pypi.org/project/django-filter/ can be used to handle validation of query params